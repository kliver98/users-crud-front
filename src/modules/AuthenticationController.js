import axios from 'axios';

const baseAPIauth = process.env.VUE_APP_BASE_URL_API_AUTH

export const AuthenticationController = {
    authenticate(credentials) {
        return axios({
            method: 'POST',
            baseURL: baseAPIauth,
            url: 'users/auth/',
            data: credentials,
        })
    }
}
