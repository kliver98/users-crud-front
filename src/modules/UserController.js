import axios from 'axios';

const baseAPIuser = process.env.VUE_APP_BASE_URL_API_USER

export const UserController = {
    create(user) {
        return axios({
            method: 'POST',
            baseURL: baseAPIuser,
            url: 'users',
            data: user,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: baseAPIuser,
            url: 'users'
        });
    },
    delete(id) {
        return axios({
            method: 'DELETE',
            baseURL: baseAPIuser,
            url: `users/${id}`,
        });
    },
    getUser(id) {
        return axios({
            method: 'GET',
            baseURL: baseAPIuser,
            url: `users/${id}`,
        });
    },
    updateUser(id, user) {
        return axios({
            method: 'PUT',
            baseURL: baseAPIuser,
            url: `users/${id}`,
            data: user,
        });
    }
}
