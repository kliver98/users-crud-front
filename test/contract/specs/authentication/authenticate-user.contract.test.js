import { provider } from '../../config/init-pact-authentication';
import { AuthenticationController as controller } from '../../../../src/modules/AuthenticationController';
import { Matchers } from '@pact-foundation/pact';

const credentials = {
    username:"admin",
    password:"password"
}

describe('Given an authentication service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to authenticate an user is made', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'check credentials of user that are correct',
                uponReceiving: 'a request to authenticate an user',
                withRequest: {
                    method: 'POST',
                    path: '/api/latest/users/auth/',
                    body: credentials
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        username: Matchers.string("admin"),
                    }, {min: 1})
                }
            });
        });

        test('Then it should return the right data', async() => {
            console.log('<<<>>>>')
            const response = await controller.authenticate(credentials);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
