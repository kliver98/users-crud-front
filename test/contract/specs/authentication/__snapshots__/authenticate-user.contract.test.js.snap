// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an authentication service When a request to authenticate an user is made Then it should return the right data 1`] = `
Array [
  Object {
    "username": "admin",
  },
]
`;
